src/app/mods/health-bar/health-bar.ts

```diff
@@ -1,153 +1,148 @@
+            margin-top: 10px;
+        }
 
-            this.shortcutsHelper.bind(this.params.health_bar_shortcut, () => {
-                this.barContainer.toggle();
-            });
+        .lifeBar {
+            transition-duration: 300ms;
+            height: 100%;
+            width: 100%;
+            background-color: #333;
         }
-    }
 
+        .lifePointsText {
+            font-size: 12px;
+            position: absolute;
+            width: 80px;
+            color: white;
+            text-shadow: 0 0 5px rgba(0, 0, 0, 0.9);
+            transition-duration: 500ms;
+            margin-top: 16px;
+            margin-left: 2px;
+        }`;
 
-    private removeOnDeath(): void {
-        this.on(this.wGame.gui, 'GameActionFightDeathMessage', (e: any) => {
-            try {
-                this.barContainer.destroyBar(e.targetId);
-            } catch (ex) {
-                Logger.error(ex);
-            }
-        });
+        this.wGame.document.getElementsByTagName('head')[0].appendChild(healthbarCss);
+
+        this.createHealthBars();
+        this.showHealthBars();
+
+        let show = () => { this.createHealthBars(); this.showHealthBars(); }
+        this.on(this.wGame.gui,'GameFightOptionStateUpdateMessage',show);
+        this.on(this.wGame.dofus.connectionManager, 'GameFightTurnStartMessage',show);
+        this.on(this.wGame.dofus.connectionManager, 'GameFightTurnEndMessage',show);
+
+        let destroy = () => {  this.destoryHealthBars();   }
+        this.on(this.wGame.dofus.connectionManager, 'GameFightLeaveMessage',destroy);
+        this.on(this.wGame.dofus.connectionManager, 'GameFightEndMessage',destroy);
+
+        this.on(this.wGame.gui, 'GameActionFightDeathMessage', (e: any) => { this.destroyHealthBar(e.targetId); });
     }
 
-    private setFightStart(): void {
-        this.on(this.wGame.dofus.connectionManager, 'GameFightStartingMessage', () => {
-            try {
-                this.fightJustStarted = true;
-            } catch (ex) {
-                Logger.error(ex);
-            }
-        });
+    private createHealthBars(){
+        if (this.wGame.document.getElementById('lifeBars') != null) return;
+        
+        this.container = document.createElement('div');
+        this.container.id = 'lifeBars';
+        this.container.className = 'lifeBarsContainer';
+        this.container.style.visibility = '';
+
+        this.wGame.foreground.rootElement.appendChild(this.container);
     }
 
-    private displayOnStart(): void {
-        this.on(this.wGame.dofus.connectionManager, 'GameFightNewRoundMessage', () => {
-            try {
-                if (this.fightJustStarted) {
-                    this.fightJustStarted = false;
-                    this.barContainer.fightStarted();
-                }
-            } catch (ex) {
-                Logger.error(ex);
-            }
-        });
+    private showHealthBars(){
+        if (this.container.style.visibility != "visible"){
```

Comment:
This implies the style HAS to be set with JS. Instead, use `window.getComputedStyle(this.container).visibility`.

Reply:
this property is always visible. not what I want

Reply:
The code I provided gives the actual value of the property. The code you use gives the value provided in the "style" attribute of the tag (which can be overridden with a "!important" annotation).